core/src/autogluon/core/calibrate/temperature_scaling.py

```diff
@@ -24,14 +24,23 @@ def tune_temperature_scaling(y_val_probs: np.ndarray, y_val: np.ndarray, init_va
         The initial learning rate
 
     Return:
-    float: The temperature scaling term
+    float: The temperature scaling term, returns None if infinity found in logits.
     '''
     try_import_torch()
     import torch
 
+    # This is required to avoid error when passing np.uint16 to torch.tensor. This can occur if >255 classes (Dionis)
+    y_val = y_val.astype('int')
+
     y_val_tensor = torch.tensor(y_val)
     temperature_param = torch.nn.Parameter(torch.ones(1).fill_(init_val))
     logits = torch.tensor(np.log(y_val_probs))
+
+    # TODO: Could alternatively add epsilon to y_val_probs in order to avoid.
+    is_invalid = torch.isinf(logits).any().tolist()
```

Comment:
Torch used to be optional dependency. If this is on by default, then we need to move it into required dependencies. This would have some deployment size implications (torch library is big). Do we need torch here or scikit methods can be used instead?

Reply:
I don't have context to say if torch is necessary for this. It probably isn't. Regardless, this is not actually a required dependency. If torch is not installed then calibration is simply skipped. Refer to line 882.